### ISSUE TITLE
Including Identity User doesn't work

### ISSUE THREAD

**Issue by SaphuA** [dotnet/efcore#3313]
Somehow EF isn't including my Identity User in results. I've tested this in a newly created default web project (beta 7).

Here's my model:

``` C#
namespace WebApplicationTest.Models
{
    public class ApplicationDbContext : IdentityDbContext<ApplicationUser>
    {
        public DbSet<Car> Cars { get; set; }
        public DbSet<City> Cities { get; set; }
    }

    public class ApplicationUser : IdentityUser
    {
        public Guid? CityId { get; set; }
        public virtual ICollection<Car> Cars { get; set; }
        public virtual City City { get; set; }
    }

    public class Car
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        public Guid UserId { get; set; }
        [Required]
        public Guid CityId { get; set; }
        public virtual City City { get; set; }
        public virtual ApplicationUser User { get; set; }
    }

    public class City
    {
        [Key]
        public Guid Id { get; set; }
        [Required]
        public Guid UserId { get; set; }
        public virtual ICollection<ApplicationUser> Users { get; set; }
        public virtual ICollection<Car> Cars { get; set; }
    }
}
```

These are my queries and the results I'm getting. I'm 100% positive that, at least according to the data, the User-results shouldn't be null.

``` C#
var car = applicationDbContext.Cars.Include(c => c.User).Include(c => c.City).Include(c => c.City.Users).First();
var city = applicationDbContext.Cities.Include(c => c.Cars).Include(c => c.Users).First();

var a1 = car.User; // Null
var a2 = car.City; // Not null
var a3 = car.City.Users.FirstOrDefault(); // Null
var a4 = city.Users.FirstOrDefault(); // Null
var a5 = city.Cars.FirstOrDefault(); // Not null
```

**Comment by natemcmaster** [comment]
EF requires that you map relationship types before `.Include` can function correctly. This can be done from the `OnModelCreating` override or (in some cases) using data annotations.

I recommend reading [this answer on StackOverflow](http://stackoverflow.com/a/30309381/2526265) as docs are not yet ready. (NB: the names of methods have changed since this answer).

For example,
```c#
protected override void OnModelCreating(ModelBuilder modelBuilder)
{
    modelBuilder.Entity<City>(entity =>
    {
        entity.HasMany(c => c.Users).WithOne(u => u.City).ForeignKey(u=> u.CityId);
    });
}
```

**Comment by SaphuA** [comment]
@natemcmaster 

Whoops I failed to mention that I was using migrations to generate everything for me.

I've done some testing and I've finally figured out the issue.

IdentityUser uses a string type as the primary key, even though it stores Guids in that field.

Once I changed the UserId properties to strings everything was working as expected.

I think it is a little weird that a string type is used even though a Guid is stored. Not sure if it's by design though.
